aspnetcore/mvc/views/razor.md

```diff
@@ -568,6 +568,25 @@ The code renders the following HTML:
 
 ::: moniker-end
 
+### @attribute
+
+The `@attribute` directive adds the given attribute onto the class of the generated view.
+
+```cshtml
+@attribute [Authorize]
+```
+
+> [!WARNING]
+> In 3.0.0-preview6 there is a known issue where `@attribute` directives do not work in `_Imports.razor` and `_ViewImports.cshtml`. This will be fixed in 3.0.0-preview7.
+
+### @namespace
+
+The `@namespace` directive sets the namespace of the class of the generated view.
```

Comment:
@ajaybhargavb do you think it's worth adding a note here on how this impacts Component inclusion similar to `@using`?

I'm on the fence.

Reply:
We don't have to mention anything about components specifically because this behaves the same as it does for non-component apps. But I think it might be worth mentioning that this will also work on imports and in which case it will set the base namespace for the related files.

Reply:
So you're saying if I set the namespace of component A, then have it import component B, component B's namespace will be set to the same as Component A's? What happens if Component C also imports B and has a different namespace set?

Reply:
if `MyApp/Pages/_ViewImports.cshtml` contains the following,
```cshtml
@namespace Hello.World
```

then the namespace will be set as follows,
`MyApp/Pages/Index.cshtml` => `Hello.World`
`MyApp/Pages/About.cshtml` => `Hello.World`
`MyApp/Pages/MorePages/Bar.cshtml` => `Hello.World.MorePages`

Reply:
And if for example, `MyApp/Pages/Index.cshtml` also contains a `@namespace` set, then that preferred. Basically the `@namespace` directive closest to the file wins.

Reply:
... also note that there is namespace guidance at **Line 49** of the components topic that might require attention ...

https://github.com/aspnet/AspNetCore.Docs/edit/master/aspnetcore/blazor/components.md

Reply:
Is `MyApp/Pages/MorePages/Bar.cshtml => Hello.World.MorePages` the real behavior in that situation? How does it know that two levels is the appropriate amount to trim down? Because if the namespace directive is on `MyApp/Pages/Shared/_Layout.cshtml` instead I feel like the desired behavior is unclear. What happens in the event of a tie? Say `MyApp/Pages/_Header.cshtml` and `MyApp/Pages/_Footer.cshtml` are both imported by Index and both have different namespace directives?

Reply:
You can't import random cshtml files. Only _ViewImports.cshtml can be imported (Or _imports.razor in case of components)

Reply:
And that's applicable to both razor and blazor?

Reply:
Yes